### PR TITLE
add async job and test

### PR DESCRIPTION
**Pull request by Joannitka-pl**
Added async job for reservation timeout at offline ticket desk so two ticket desks cannot sell the same ticket at the same time.

### CODE REVIEW

File: app/concepts/reservations/use_cases/create.rb [hunk 1]
```diff
@@ -9,6 +9,11 @@ def initialize(repository: Reservations::Repository.new)
 
       def call(params:)
         repository.create(params)
+
+      if ticket_desk.category == offline
+        ReservationTimeoutJob.set(wait: 10.minutes).perform_later(reservation)
```
Comment (tobiaszwaszak) [inline app/concepts/reservations/use_cases/create.rb:14]: you don't have `reservation`
if you want to use it you have to assign it `reservation = repository.create(params)`
but in this case, you will not return the reservation (you have to add it in last line of method call)
and, you don't have `ticket_desk here`

File: app/jobs/reservation_timeout_job.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+class ReservationTimeoutJob < ApplicationJob
+  queue_as :default
+
+  def perform(reservation)
+    Reservations::UseCases::Destroy.new.call(id: params[:id])
```
Comment (tobiaszwaszak) [inline app/jobs/reservation_timeout_job.rb:5]: you don't have params here

File: config/routes.rb [hunk 3]
```diff
@@ -1,4 +1,10 @@
 Rails.application.routes.draw do
   resources :halls, :seances, :movies, :tickets, 
             :reservations, :clients, :ticket_desks, :seats
+
+  require 'sidekiq/web'
```
Comment (tobiaszwaszak) [inline config/routes.rb:5]: require should be on the top

File: spec/jobs/reservation_timeout_job_spec.rb [hunk 4]
```diff
@@ -0,0 +1,13 @@
+require "rails_helper"
+require 'spec_helper'
+
+RSpec.describe 'ReservationTimeout', :type => :job do
+  describe "#perform_later" do
+    it "destroys reservation at offline ticket desk after 10 minutes unless paid" do
+      ActiveJob::Base.queue_adapter = :test
+      expect {
+        ReservationTimeoutJob.perform_later('reservation')
```
Comment (tobiaszwaszak) [inline spec/jobs/reservation_timeout_job_spec.rb:9]: something like that you should test in `concepts/reservations/use_cases/create_spec.rb`
in this test, you could do something like

```
it do
    described_class.new.perform(reservation)
    expect(Reservations::UseCases::Destroy).to have_received(:call)
 end
```

File: app/concepts/reservations/use_cases/create.rb (hunk 1 above)
Comment (Joannitka-pl) [reply app/concepts/reservations/use_cases/create.rb]: after consultation with @wmaciejak we have come to the conclusion to put `reservation = repository.create(params)` in the beginning of the method `call` to be able to access reservation.id.

File: config/routes.rb (hunk 3 above)
Comment (Joannitka-pl) [reply config/routes.rb]: - [ ]

File: spec/jobs/reservation_timeout_job_spec.rb (hunk 4 above)
Comment (Joannitka-pl) [reply spec/jobs/reservation_timeout_job_spec.rb]: Unfortunately this test doesn't pass. Not sure why? The error message says: undefined local variable or method 'reservation_id' for... create_spec.rb.7.

Comment (tobiaszwaszak) [reply spec/jobs/reservation_timeout_job_spec.rb]: hard to say, you didn't push changes

Comment (Joannitka-pl) [reply spec/jobs/reservation_timeout_job_spec.rb]: They're here:
https://github.com/Joannitka-pl/monte-cinema/pull/6/commits/908e4c5818b5d70ad985c3975eff169284b5afcc 
I can see this commit below our discussion.

Comment (tobiaszwaszak) [reply spec/jobs/reservation_timeout_job_spec.rb]: you have to assign something as a reservation_id
or create a reservation and use reservation.id

Comment (Joannitka-pl) [reply spec/jobs/reservation_timeout_job_spec.rb]: still not working when testing with Rspec...

Comment (Joannitka-pl) [reply spec/jobs/reservation_timeout_job_spec.rb]: Screenshot 2021-06-02 at 16 05 24

Comment (wmaciejak) [reply spec/jobs/reservation_timeout_job_spec.rb]: Could you please push your latest changes to this branch?

Comment (Joannitka-pl) [reply spec/jobs/reservation_timeout_job_spec.rb]: I have pushed it already and can see corresponding commit below our chat. 
Here: https://github.com/Joannitka-pl/monte-cinema/pull/6/commits/01a63f24a210391b3226c45ab3baa85b0a45bf2d 
Am I doing it wrong?

Comment (wmaciejak) [reply spec/jobs/reservation_timeout_job_spec.rb]: I missed it probably or I looked into wrong files, sorry :)

Comment (Joannitka-pl) [reply spec/jobs/reservation_timeout_job_spec.rb]: @wmaciejak Hey, have you had a chance to look into it?

**Review by wmaciejak — CHANGES_REQUESTED** [review]

File: spec/jobs/reservation_timeout_job_spec.rb (hunk 4 above)
Comment (wmaciejak) [reply spec/jobs/reservation_timeout_job_spec.rb]: Please check this comment
https://github.com/Joannitka-pl/monte-cinema/pull/6#discussion_r644127950

**Pull request closed without merging**